File: _code_examples/kamon-core-examples/src/main/java/kamon/examples/java/TraceContextManipulation.java
package kamon.examples.java;


import kamon.Kamon;
import kamon.trace.TraceContext;

public class TraceContextManipulation {
  public static void main(String[] args) throws InterruptedException {
    Kamon.start();

    // tag:creating-a-trace-context:start
    final TraceContext newContext = Kamon.tracer().newContext("test-trace");

    Thread.sleep(3000);
    newContext.finish();
    // tag:creating-a-trace-context:end
  }
}

File: _code_examples/kamon-core-examples/src/main/scala/kamon/examples/scala/TraceContextManipulation.scala
package io.kamon.examples

import kamon.Kamon

object TraceContextManipulation extends App {
  Kamon.start()

  // tag:creating-a-trace-context:start
  val newContext = Kamon.tracer.newContext("test-trace")

  Thread.sleep(3000)
  newContext.finish()
  // tag:creating-a-trace-context:end

}